### PR TITLE
feat: Allow multiple org owners

### PR DESCRIPTION
**Pull request by xrdt**
## Problem

This should reduce the number of support tickets we get asking us to transfer org ownership after the original owner has left.

Changes: 
* Allow multiple org owners 
* Update the UI to allow making other users owners 
* Update the invite UI to allow selecting the level at which you want to invite a new user (defaults to member)

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Comment by xrdt** [comment]
Demo: 

https://github.com/PostHog/posthog/assets/21014901/f2246961-0dcd-46d8-a096-78dcd0209b50

File: frontend/src/scenes/settings/organization/Members.tsx [hunk 1]
```diff
@@ -69,11 +69,13 @@ function ActionsComponent(_: any, member: OrganizationMemberType): JSX.Element |
...
-                                            title: `Transfer organization ownership to ${fullName(member.user)}?`,
-                                            description: `You will no longer be the owner of ${user.organization?.name}. After the transfer you will become an administrator.`,
+                                            title: `Add additional owner to ${user.organization?.name}?`,
+                                            description: `Please confirm that you would like to make ${fullName(
+                                                member.user
+                                            )} an owner of ${user.organization?.name}.`,
                                             primaryButton: {
                                                 status: 'danger',
-                                                children: 'Transfer Ownership',
+                                                children: `Make ${fullName(member.user)} owner`,
```
Comment (zlwaterfield) [inline frontend/src/scenes/settings/organization/Members.tsx:78]: Does it sound better as `Make ${fullName(member.user)} an owner`?

File: frontend/src/scenes/settings/organization/InviteModal.tsx [hunk 2]
```diff
@@ -97,18 +112,30 @@ export function InviteRow({ index, isDeletable }: { index: number; isDeletable:
                         }}
                     />
                 )}
-                {isDeletable && (
-                    <LemonButton icon={<IconTrash />} status="danger" onClick={() => deleteInviteAtIndex(index)} />
-                )}
             </div>
+            <div className="flex-1 flex gap-1 items-center justify-between">
+                <LemonSelect
```
Comment (zlwaterfield) [inline frontend/src/scenes/settings/organization/InviteModal.tsx:117]: What happens if a member is inviting someone? Should we just hide this?

**Review by zlwaterfield — CHANGES_REQUESTED** [review]
Awesome work here! Excited to have this :) 

Can we add some tests here since this is such a sensitive area around org ownership.

I'd imagine both stories for visual regress and validation for API would be needed.

File: frontend/src/scenes/settings/organization/Members.tsx (hunk 1 above)
Comment (xrdt) [reply frontend/src/scenes/settings/organization/Members.tsx:78]: Yes, good point.

File: frontend/src/scenes/settings/organization/InviteModal.tsx (hunk 2 above)
Comment (xrdt) [reply frontend/src/scenes/settings/organization/InviteModal.tsx:117]: Will hide if there is only one allowed option.

**Comment by xrdt** [comment]
Thanks for the reviews, @zlwaterfield , have updated existing tests and added a storybook mock.

File: frontend/src/scenes/settings/organization/Members.tsx (hunk 1 above)
Comment (zlwaterfield) [reply frontend/src/scenes/settings/organization/Members.tsx:78]: Didn't see this get updated

**Review by zlwaterfield — APPROVED** [review]
Just one comment about the button copy. 

Flyby: I think we should add more UI tests here for the modal being open from the perspective of each role type to make sure the dropdown appears as expected but not a blocker, feel free to do that at another time.

**Comment by xrdt** [comment]
Some new modifications: 
* Removed the pending invites from the invite modal. You can see and delete these on the page itself so it's not clear that having an additional display for them in the modal is useful. They were also cluttering up the alignment of the modal. 
* Added the level a user is invited as to the pending invites table. 
* Fixed the cancel modal layout. 
* Adjusted LemonModal header right padding so it doesn't overlap with the close button. 
* Added some more storybook mocks to showcase the different options depending on the current user's organization membership level.

File: .vscode/launch.json [hunk 3]
```diff
@@ -71,7 +71,7 @@
                 "DATABASE_URL": "postgres://posthog:posthog@localhost:5432/posthog",
                 "SKIP_SERVICE_VERSION_REQUIREMENTS": "1",
                 "PRINT_SQL": "1",
-                "BILLING_SERVICE_URL": "https://billing.dev.posthog.dev",
+                "BILLING_SERVICE_URL": "http://localhost:8100", // "https://billing.dev.posthog.dev",
```
Comment (zlwaterfield) [inline .vscode/launch.json:74]: Can you put this back?

File: frontend/src/scenes/settings/organization/InviteModal.tsx [hunk 4]
```diff
@@ -124,69 +165,11 @@ export function InviteTeamMatesComponent(): JSX.Element {
             )}
             <div className="space-y-2">
                 <div className="flex gap-2">
-                    <b className="flex-1">Email address</b>
+                    <b className="flex-2">Email address</b>
                     <b className="flex-1">{preflight?.email_service_available ? 'Name (optional)' : 'Invite link'}</b>
+                    {allowedLevelsOptions.length > 1 && <b className="flex-1">Level</b>}
                 </div>
 
-                {invitesReversed.map((invite: OrganizationInviteType) => {
```
Comment (zlwaterfield) [inline frontend/src/scenes/settings/organization/InviteModal.tsx:131]: To confirm this is just removing existing invites from the modal? Works for me.

File: test.env [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export REDIS_URL='redis://localhost'
```
Comment (zlwaterfield) [inline test.env:1]: Remove this file?

File: frontend/src/scenes/settings/organization/InviteModal.tsx (hunk 4 above)
Comment (xrdt) [reply frontend/src/scenes/settings/organization/InviteModal.tsx:131]: Yeah, I'm removing existing invites. I thought it looked cluttered and didn't think it added anything beyond the pending invites table already on the settings page.

**Pull request merged**